### ISSUE TITLE
Magento 2.2.8 Can't save products after updating

### ISSUE THREAD

**Issue by Ctucker9233** [magento/magento2#22274]
After updating to 2.2.8 I can't save a product after updating it. Clicking the save and close button appears to do nothing. No error messages appear.

### Preconditions (*)
Magento 2.2.8
Some third party extensions installed although they have all confirmed for me that it has nothing to do with their extensions.

### Steps to reproduce (*)

1. Make a required field in advanced inventory
2. Open a product to edit it
3. Make changes
4. Click save and close

### Expected result (*)
1. Product saves successfully

### Actual result (*)
1. Nothing happens when the button is clicked. No error messages appear to indicate missing information.

After some useful comments, the cause of this is an error that is thrown in the advanced inventory options. This remains invisible unless the user manually expands the advanced inventory options. An error flag should appear that indicates the location of the error.

**Comment by Ctucker9233** [comment]
@magento-engcom-team Give me 2.2-develop instance with sample data

**Comment by magento-engcom-team** [comment]
Hi @Ctucker9233. Thank you for your request. I'm working on Magento 2.2-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @Ctucker9233, here is your [Magento instance](https://i-22274-2-2-develop.instances.magento-community.engineering).
Admin access: [https://i-22274-2-2-develop.instances.magento-community.engineering/admin](https://i-22274-2-2-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by Ctucker9233** [comment]
@magento-engcom-team how can I get some sample data in this instance? Without it, I can't confirm. Adding new products and then later editing those works just fine. It is only products that existed prior to 2.2.8 that are affected.

**Comment by mad-develop** [comment]
@Ctucker9233 i expierienced the same problem. For me there where custom settings in the "**Advanced Inventory**" ("**Maximum Qty Allowed in Shopping Cart**" was 0 so the form validation failed.) Cost me several hours to figure this out. Maybe its the same for you.

**Comment by nadeefit6** [comment]
This is probably a missing required product attribute or field in a different tab.
Other than that this issue cannot be reproduced in 2.2.7.

**Comment by steinkopffp** [comment]
I faced a similar issue is which was caused by a change on the attribute **Maximum Qty Allowed in Shopping Cart**. Push the save button and look in modals like **Advanced Inventory** and search for an error inside modals.

**Comment by Ctucker9233** [comment]
@mad-develop at @steinkopffp I will look at that right now. Thank you!

**Comment by Ctucker9233** [comment]
@mad-develop @steinkopffp that was it! Now all we need is for @magento-engcom-team to add an error message that validates advanced inventory so that more people like me don't spend hours banging their heads against a wall.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99265`, `MAGETWO-99266` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by Krishnan369** [comment]
"I am working on this at #dmcdindia19”

**Comment by magento-engcom-team** [comment]
@Krishnan369 thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by vidhyasaharJ** [comment]
Give me 2.2.8-develop instance with sample data

**Comment by vidhyasaharJ** [comment]
Give me 2.2.8-develop instance with sample data

**Comment by vidhyasaharJ** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @vidhyasaharJ. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @vidhyasaharJ, here is your [Magento instance](https://i-22274-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-22274-2-3-develop.instances.magento-community.engineering/admin](https://i-22274-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by vidhyasaharJ** [comment]
give me 2.2.8-develop instance with sample data

**Comment by vidhyasaharJ** [comment]
@magento-engcom-team give me 2.2.8-develop instance with sample data

**Comment by magento-engcom-team** [comment]
Hi @vidhyasaharJ. Thank you for your request. I'm working on Magento 2.2.8-develop instance for you

**Comment by archonkulis** [comment]
I also had this problem.

The problem was that Magento tried to focus on the field that was hidden and validation inside `vendor/magento/module-ui/view/base/web/js/form/form.js save()` function didn't pass and it ended up inside `focusInvalid()` function. I console logged `invalidField`and in one of the objects parameters I found the field class `max_sale_qty` and then found it in the html document, removed `display: none` parameter and there was an error "Please enter a number greater than 0 in this field." (https://prnt.sc/nlvlf4)

Looks like client imported some products and changed `use_config_max_sale_qty` to 1 or something like that.

So for me, this doesn't seem to be related to upgrading, but hard to tell for 100%.

In short - the problem seems to be that Magento should remove `display:none` from the field that validation failed so that clients can see the error message.

Cheers

**Comment by mallikaverma** [comment]
@magento-engcom-team Give me 2.2.8-develop instance with sample data

**Comment by magento-engcom-team** [comment]
Hi @mallikaverma. Thank you for your request. I'm working on Magento 2.2.8-develop instance for you

**Comment by mallikaverma** [comment]
#imagine2019

**Comment by itakiyev** [comment]
> I faced a similar issue is which was caused by a change on the attribute **Maximum Qty Allowed in Shopping Cart**. Push the save button and look in modals like **Advanced Inventory** and search for an error inside modals.

My goddess, appreciate your answer, spend quite amount of time lmao

**Comment by MaxSouza** [comment]
Hi everyone,

As this is configuration,  I'll close the issue.

**Comment by Ctucker9233** [comment]
@MaxSouza What do you mean this is a configuration problem? Can this configuration be changed?

**Comment by MaxSouza** [comment]
@Ctucker9233 Sorry, there is an error because the validation field does not display when you tried save product.

**Comment by Ctucker9233** [comment]
@MaxSouza Can you add the tags back that the Magento bot removed?

**Comment by magento-engcom-team** [comment]
@MaxSouza Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by MaxSouza** [comment]
@Ctucker9233 Done.

**Comment by Ctucker9233** [comment]
@MaxSouza Thank you

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-21821` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @Ctucker9233. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. 

The fix will be available with the upcoming `2.3.5` release.
